src/main/java/com/siit/hospital_manager/controller/DoctorController.java

```diff
@@ -0,0 +1,130 @@
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+
+import java.util.List;
+
+import static com.siit.hospital_manager.util.AuthUtils.isAdmin;
+import static com.siit.hospital_manager.util.AuthUtils.isPatient;
+
+
+@Controller
+@RequiredArgsConstructor
+@RequestMapping("/mvc/doctor")
+public class DoctorController {
+
+    private final DoctorService doctorService;
+    private final SpecialisationService specialisationService;
+
+    @GetMapping("/viewAll")
+    public String viewAll(Model model, Authentication authentication){
+        List<DoctorDto> doctorsList = doctorService.findAll();
+        model.addAttribute("doctors", doctorsList);
+        if (isAdmin(authentication)) {
+            List<SpecialisationDto> specialisationList = specialisationService.findAll();
+            model.addAttribute("specialisations", specialisationList);
+            return "admin/viewAllDoctors";
+        }
+        else {
+            return "doctor/viewAll";
+        }
+    }
+
+    @GetMapping("/viewDoctorsBySpecialisation/{specialisationId}")
+    public String viewDoctorsBySpecialisation(Model model, @PathVariable("specialisationId") Integer specialisationId, Authentication authentication) {
+        List<DoctorDto> doctorsList = doctorService.findAllBySpecialisation(specialisationService.findSpecialisationById(specialisationId));
+        if (doctorsList.isEmpty() && isAdmin(authentication)) {
+            return "redirect:/mvc/doctor/createDoctor/{specialisationId}";
+        }
+        else if (isAdmin(authentication)) {
+            model.addAttribute("doctors", doctorsList);
+            return "admin/viewDoctorsBySpecialisation";
+        }
+        else {
+            List<DoctorDto> activeDoctorsList = doctorService.findAllActiveDoctorsBySpecialisation(specialisationService.findSpecialisationById(specialisationId));
+            model.addAttribute("doctors", activeDoctorsList);
+            return "doctor/viewDoctorsBySpecialisation";
+        }
+    }
+
+    @GetMapping("/viewDoctorProfile/{id}")
+    public String viewDoctorProfile(Model model, @PathVariable("id") Integer id, Authentication authentication) {
+        if (isAdmin(authentication)) {
+            DoctorDto doctorDto = doctorService.findById(id);
+            model.addAttribute("doctor", doctorDto);
+            List<SpecialisationDto> specialisationList = specialisationService.findAll();
+            model.addAttribute("specialisations", specialisationList);
+            return "admin/viewDoctorProfile";
+        } else {
+            DoctorDto doctor = doctorService.findActiveById(id);
+            model.addAttribute("doctor", doctor);
+            if (isPatient(authentication)) return "patient/viewDoctorProfile";
+            else return "doctor/viewDoctorProfile";
```

Comment:
all these nested if statements make the code harder to read. you could separate methods with different logic in the service